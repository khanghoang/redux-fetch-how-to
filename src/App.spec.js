const nodeFetch = require('node-fetch');
global.fetch = nodeFetch;

import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Root from './root';
import usersData from './users';
import nock from 'nock';

nock('https://jsonplaceholder.typicode.com')
  .get('/users')
  .reply(200, JSON.stringify(usersData));

describe('Root', () => {
  it('renders list of users', done => {
    const users = mount(<Root />).find("#users");
    setTimeout(function() {
      expect(toJson(users)).toMatchSnapshot();
      done();
    }, 10);
  });
});
