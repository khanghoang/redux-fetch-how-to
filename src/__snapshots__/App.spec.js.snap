// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root renders list of users 1`] = `
<div
  id="users"
>
  <Username
    username="Leanne Graham"
  >
    <p>
      Leanne Graham
    </p>
  </Username>
  <Username
    username="Ervin Howell"
  >
    <p>
      Ervin Howell
    </p>
  </Username>
  <Username
    username="Clementine Bauch"
  >
    <p>
      Clementine Bauch
    </p>
  </Username>
  <Username
    username="Patricia Lebsack"
  >
    <p>
      Patricia Lebsack
    </p>
  </Username>
  <Username
    username="Chelsey Dietrich"
  >
    <p>
      Chelsey Dietrich
    </p>
  </Username>
  <Username
    username="Mrs. Dennis Schulist"
  >
    <p>
      Mrs. Dennis Schulist
    </p>
  </Username>
  <Username
    username="Kurtis Weissnat"
  >
    <p>
      Kurtis Weissnat
    </p>
  </Username>
  <Username
    username="Nicholas Runolfsdottir V"
  >
    <p>
      Nicholas Runolfsdottir V
    </p>
  </Username>
  <Username
    username="Glenna Reichert"
  >
    <p>
      Glenna Reichert
    </p>
  </Username>
  <Username
    username="Clementina DuBuque"
  >
    <p>
      Clementina DuBuque
    </p>
  </Username>
</div>
`;
